### PR TITLE
[HxSidebar] HxSidebarItem ContentTemplate #212

### PR DESCRIPTION
**Pull request by Harvey1214**
A new ContentTemplate parameter has been added to HxSidebarItem in order to allow custom content such as a HxBadge to be added to the items' contents.

ChildContent RenderFragment parameter (intended for sub-items) has been renamed to SubItems to comply with the library's conventions. The ChildContent parameter is currently still in function in order to continually prepare users for the breaking change.

Not sure about the ContentTemplate naming, happy to hear suggestions, thanks.

### CODE REVIEW

File: Havit.Blazor.Components.Web.Bootstrap/Navigation/HxSidebarItem.razor [hunk 1]
```diff
@@ -12,13 +12,21 @@
...
+                            }
+                            
+                            <span class="@CssClassHelper.Combine("text", Icon == null ? null : "ms-3")">
+                                @Text
+                            </span>
+                        }
+                        else
+                        {
+                            @ContentTemplate
                         }
```
Comment (hakenr) [inline Havit.Blazor.Components.Web.Bootstrap/Navigation/HxSidebarItem.razor:29]: We should not replace the current item with the `ContentTemplate`, but add it to the other options (similar patter is used in other components where it makes sense).
I would go for order (chevron icon just for expandable items)
```
  @if ... <HxIcon />
  @ContentTemplate
  <span ...>@Text</span>
  <HxIcon Icon="@BootstrapIcon.ChevronRight" CssClass="expand-icon" />
```

File: Havit.Blazor.Components.Web.Bootstrap/Navigation/HxSidebarItem.razor [hunk 2]
```diff
@@ -36,22 +44,31 @@
...
+                        }
+                        else
                         {
-                            <HxIcon Icon="Icon" />
+                            @ContentTemplate
                         }
-                        <span class="text">
-                            @Text
-                        </span>
-                        <HxIcon Icon="@BootstrapIcon.ChevronRight" CssClass="expand-icon" />
```
Comment (hakenr) [inline Havit.Blazor.Components.Web.Bootstrap/Navigation/HxSidebarItem.razor:46]: We should not drop the chevron-icon for version with `@ContentTemplate`, see above.

File: Havit.Blazor.Components.Web.Bootstrap/Navigation/HxSidebarItem.razor.cs [hunk 3]
```diff
@@ -48,10 +48,26 @@ public partial class HxSidebarItem : IAsyncDisposable
...
+		/// <summary>
+		/// Use the <see cref="SubItems"/> parameter instead as <see cref="ChildContent"/> is deprecated.
+		/// </summary>
+		[Parameter]
+		[Obsolete("ChildContent is obsolete, please use the SubItems parameter instead.")]
+		public RenderFragment ChildContent
+		{
+			set => SubItems = value;
+			get => SubItems;
+		}
```
Comment (hakenr) [inline Havit.Blazor.Components.Web.Bootstrap/Navigation/HxSidebarItem.razor.cs:70]: Keep the parameter as `ChildContent` without the move to `SubItems`. I found this pattern in other libraries and makes sense for me. We might consolidate our other components later.

**Review by hakenr — CHANGES_REQUESTED** [review]

**Comment by hakenr** [comment]
+ [ ] add demo with `ContentTemplate` usage, e.g. the badges requested.

**Comment by Harvey1214** [comment]
@hakenr Do we want to automatically hide the ContentTemplate when a sidebar is collapsed? Currently, it is always rendered, which leaves it up to the user to handle the hiding that I don't find ideal. Please let me know what you think, thanks.

**Comment by hakenr** [comment]
Ahh, I see. The visibility of the item-content is currently bound to `.text` class. If it is a custom class, let's rename it to `.hx-sidebar-item-content` and arrange the HTML like this:
```razor
<div class="hx-sidebar-item-content ..">
  @Text
  @ContentTemplate      
</div>
```
@crdo?

**Review by hakenr — APPROVED** [review]
lgtm, the other issues will be resolved in separate ticker

**Pull request merged**